src/hffs/fs.py

```diff
@@ -153,26 +163,40 @@ def _strip_protocol(cls, path):
         protocol = cls.protocol
         if path.startswith(protocol + "://"):
             path = path[len(protocol + "://") :]
-            repo_id, *paths = path.split(":/", 1)
+            hf_id, *paths = path.split(":/", 1)
+            path = paths[0] if paths else cls.root_marker
+        # TODO: the hack below is needed to work with DuckDB, remove it as soon as it is fixed on their side
```

Comment:
What's the bug you're fixing here? Have a link on hand?

Reply:
I think the problem stems from using [`stripPrefix`](https://github.com/duckdb/duckdb/blob/d2ef69494ef54f50d8c9073fdaa5d2b8dad7c99d/tools/pythonpkg/src/include/duckdb_python/pyfilesystem.hpp#L42) instead of `fsspec`'s [`_strip_protocol`](https://github.com/fsspec/filesystem_spec/blob/35da646cf4baea2a4be43b61fbfd9a952f480c3f/fsspec/spec.py#L178), which can be overwritten to work with custom schemas.

Reply:
Alternatively, if you can provide a code snippet demonstrating the issue that would be great

Reply:
Sure, I'll create an issue!